src/components/TextField/TextField.scss

```diff
@@ -170,6 +170,22 @@ $stacking-order: (
   pointer-events: none;
 }
 
+.CharacterCount {
+  @include text-emphasis-subdued;
+  z-index: z-index(contents, $stacking-order);
+  margin-right: $backdrop-horizontal-spacing;
+  pointer-events: none;
+}
+
+.MaxCharacters {
+  color: color('red');
+}
+
+.FlexEnd {
+  align-self: flex-end;
+  margin-bottom: rem(4px);
```

Comment:
What's the purpose of setting a `margin-bottom` here? Also, this could use `spacing(extra-tight)`.